src/app/services/route-parts/route-parts.service.ts

```diff
@@ -54,6 +54,8 @@ export class RoutePartsService {
 
       return this.generateRouteParts(child, url, routeParts);
     }
+
+    return undefined;
```

Comment:
simple **return;** 
should make the same undefined in the end, isn't it?

Reply:
Because the way linter rule works is that it should either have all return values specified explicitly, or all returns be just `return` with no value. In here there is a `return this.generateRouteParts(child, url, routeParts);` above that may get returned sometimes.